docs/okeps/okep-5494-ovn-kubernetes-mcp-server.md

```diff
@@ -0,0 +1,740 @@
+abstractions to connect to the database (see discarded alternatives), it
+offers the fastest path to value for OVN-Kubernetes engineers who are already
+familiar with these tools but want to leverage LLM assistance for command
+construction and output analysis. The CLI-based approach is also optimal for
+exposing the complete troubleshooting toolkit across all layers of the stack.
+Most other discarded ideas only addressed OVSDB access and reusability
+concerns while failing to provide the holistic set of tools needed for
+comprehensive root cause analysis like running packet traces.
+
+The MCP server acts as a secure execution bridge, translating natural language
+troubleshooting requests into appropriate CLI commands.
+
+**Advantages**:
+
+* **Fastest Time to Value**: Leverages existing tools that engineers already
+  know
+* **Zero Deployment Overhead**: All required CLI binaries are already present
+  in the pods and nodes within the cluster
+* **Comprehensive Coverage**: Only approach that provides access to the
+  complete troubleshooting toolkit across all stack layers
+* **Security Controls**: Enables proper RBAC and read-only access enforcement
+* **No version compatibility issues** between the MCP server tools and the
+  cluster's installed versions when running on live cluster environments
+
+**Trade-offs**:
+
+* **Limited Reusability**: Somewhat specific to OVN-Kubernetes deployment
+  patterns and can't be reused in other layers like OVS
+* **Tool-Specific Knowledge**: Requires understanding of each CLI tool's
+  parameters and output formats
+
+This approach was selected as the optimal balance between security,
+functionality, and development effort.
+
+## Security Model and RBAC Constraints
+
+No matter how we approach this implementation, it is impossible to totally
+secure the execution of networking troubleshooting tools at this stage. Most
+networking tools require privileged access to function properly, creating
+inherent security trade-offs. The goal is therefore to minimize blast radius
+through layered controls rather than achieve perfect isolation.
+
+**Kubernetes Layer Security:**
+
+* `kubectl exec` and `oc debug node` operations require cluster-admin level
+  privileges by design
+* Alternative approach using custom debug containers
+  (`oc debug node --image=&lt;custom-image>`) with volume mounts to database
+  files reduces some attack surface but remains intrusive
+* **Mitigation**: Strict command allowlisting - expose only read operations
+  (`get`, `list`, `describe`, `logs`) while blocking any modification commands
+* **Constraint**: MCP Server service account requires elevated privileges
+  (including `pods/exec`) despite being conceptually a "troubleshooter" role
+
+For the kubernetes layer, we are going to leverage the opensource
+[kubernetes-mcp-server](https://github.com/containers/kubernetes-mcp-server).
+So whatever security posture they use for now, can be adopted.
```

Comment:
Is the expectation to reuse the same code or maintain a separate version of the code?
Support for new CRDs specific to ovnk will be needed on top of the existing kubectl command support in kubernetes-mcp-server.

Reply:
so we are not going to do any kubectl get RA or kubectl get UDN type tools- because that is like overcoming the "model not having enough context problem" - kube-mcp-server should be able to handle all the kubernetes layer calls (kubectl get crd and then it knows all the custom resources in the cluster). Anything it can't do is actually the model problem of not having enough context about an ovn-k feature.

so either we need a ovnkube-cli work that nvidia was working on OR we need to invest time in RAGing to add more context